.github/workflows/test.yml

```diff
@@ -33,7 +36,6 @@ jobs:
         - ubuntu
         - macOS
         go:
-        - 14
```

Comment:
Why was go 1.14 removed here? I think that deserves it's own PR / proposal to remove support for Go 1.14

Reply:
See run 232 in https://github.com/umarcor/cobra/actions?query=branch%3Aumarcor%2Fci%2Fcache. Enabling this feature fails with go 1.14.

Reply:
1.14 has been [end of life for awhile now](https://endoflife.date/go) so this seems fine from me. Do we have a published support policy somewhere?

Reply:
@stmcginnis see https://github.com/spf13/cobra/blob/main/CONDUCT.md#deprecation.

Reply:
Thanks! So I think as long as we follow that deprecation policy, this should be good to do.

Since that will require a major version bump according to the policy, maybe we should also think about dropping some of the other EOL Go releases and see if there are any other major changes that need to be merged before making that jump.

Reply:
Agree. Nonetheless, note that 'major' in the context of cobra is probably equivalent to 'minor' in semver semantics.

Reply:
Yeah, we should avoid doing a major version bump.  A new major version would mean a new module path: `github.com/spf13/cobra/v2`.  That means every project would need to change _all_ their Cobra _package_ import paths.  That's a big impact.

Reply:
From the `CONDUCT.md`:

> Deprecation of Go versions or dependent packages will only occur in major releases.

It's been awhile since I read that deprecation policy but I'm not sure I agree with it anymore: we should in theory be able to drop "supported" versions of our dependencies without having any breaking changes to the core functionality of cobra. People could still use cobra without issue it just wouldn't be _"supported"_.

I'm ok with the change in this PR but i think it'd be worth re-visiting the deprecation policy all together. It'll never be sustainable for us to make major releases with how fast Go deprecates it's versions.

Reply:
I have to agree. Removing support for an old, end-of-life'd, version of Go without changing any backwards functionality doesn't seem like it's worth a major version bump. It would be good to update that policy to clarify that, but I wouldn't think it's something that needs to happen before this change is made.